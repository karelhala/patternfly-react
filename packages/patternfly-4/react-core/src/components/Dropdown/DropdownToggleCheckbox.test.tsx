import * as React from 'react';
import { shallow } from 'enzyme';
import { DropdownToggleCheckbox } from './DropdownToggleCheckbox';

const props = {
  onChange: jest.fn(),
  isChecked: false
};

test('controlled', () => {
  const view = shallow(<DropdownToggleCheckbox isChecked id="check" aria-label="check" />);
  expect(view).toMatchSnapshot();
});

test('uncontrolled', () => {
  const view = shallow(<DropdownToggleCheckbox id="check" aria-label="check" />);
  expect(view).toMatchSnapshot();
});

test('with text', () => {
  const view = shallow(<DropdownToggleCheckbox id="check" isDisabled aria-label="check">Some text</DropdownToggleCheckbox>);
  expect(view).toMatchSnapshot();
});

test('isDisabled', () => {
  const view = shallow(<DropdownToggleCheckbox id="check" isDisabled aria-label="check" />);
  expect(view).toMatchSnapshot();
});

test('3rd state', () => {
  const view = shallow(<DropdownToggleCheckbox id="check" isChecked={null} aria-label="check" />);
  expect(view).toMatchSnapshot();
});

test('passing class', () => {
  const view = shallow(
    <DropdownToggleCheckbox label="label" className="class-123" id="check" isChecked aria-label="check" />
  );
  expect(view).toMatchSnapshot();
});

test('passing HTML attribute', () => {
  const view = shallow(
    <DropdownToggleCheckbox label="label" aria-labelledby="labelId" id="check" isChecked aria-label="check" />
  );
  expect(view).toMatchSnapshot();
});
