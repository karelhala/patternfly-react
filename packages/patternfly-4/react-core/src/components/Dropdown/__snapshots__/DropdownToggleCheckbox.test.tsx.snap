// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`3rd state 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    className=""
    id="check"
    isChecked={null}
    isDisabled={false}
    isValid={true}
    onChange={[Function]}
  />
</label>
`;

exports[`controlled 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    defaultChecked={true}
    disabled={false}
    id="check"
    isChecked={true}
    isDisabled={false}
    isValid={true}
    onChange={[Function]}
  />
</label>
`;

exports[`isDisabled 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    defaultChecked={null}
    disabled={true}
    id="check"
    isChecked={false}
    isDisabled={true}
    isValid={true}
    onChange={[Function]}
  />
</label>
`;

exports[`passing HTML attribute 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    aria-labelledby="labelId"
    defaultChecked={true}
    disabled={false}
    id="check"
    isChecked={true}
    isDisabled={false}
    isValid={true}
    label="label"
    onChange={[Function]}
  />
</label>
`;

exports[`passing class 1`] = `
<label
  className="pf-c-dropdown__toggle-check class-123"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    defaultChecked={true}
    disabled={false}
    id="check"
    isChecked={true}
    isDisabled={false}
    isValid={true}
    label="label"
    onChange={[Function]}
  />
</label>
`;

exports[`uncontrolled 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    defaultChecked={null}
    disabled={false}
    id="check"
    isChecked={false}
    isDisabled={false}
    isValid={true}
    onChange={[Function]}
  />
</label>
`;

exports[`with text 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <input
    aria-invalid={false}
    aria-label="check"
    defaultChecked={null}
    disabled={true}
    id="check"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    aria-hidden="true"
    className="pf-c-dropdown__toggle-text"
    id="check-text"
  >
    Some text
  </span>
</label>
`;
